gpflow/models/model.py

```diff
@@ -85,53 +80,14 @@ def maximum_likelihood_objective(self, *args, **kwargs) -> tf.Tensor:
         raise NotImplementedError
 
 
-class BayesianModelStoringData(BayesianModel):
-    """
-    Base class for models that encapsulate their data and store it as
-    self.data; training_loss does not take any arguments.
-    """
-
-    def training_loss(self):
-        return self._training_loss()
-
-    def training_loss_closure(self, jit=True) -> Callable[[], tf.Tensor]:
-        if jit:
-            return tf.function(self.training_loss)
-        else:
-            return self.training_loss
-
-
-class BayesianModelNotStoringData(BayesianModel):
+class ExternalDataBayesianModel(BayesianModel, ExternalDataTrainingInterface):
```

Comment:
this appears only to be used in the notebooks. do we want to have something in the codebase that's only there for a single notebook?

Reply:
do we expect users to use it?

Reply:
yes.